### PR TITLE
Refactor allowed domain in workspace

### PR DESCRIPTION
**Pull request by flvndvd**
This PR refactors the current system for managing the workspace whitelisted domain, which facilitates the process of new users joining the workspace via an invite. This update serves as groundwork for the suggested approach in https://github.com/dust-tt/dust/issues/3154.

Following the changes in this PR, users will no longer have the option to select their whitelisted domain. Instead, the platform will automatically adopt the email domain of the current admin user, provided it is not marked as a disposable or low-quality domain according to HubSpot's criteria.

This PR represents the initial phase of the process. The next step will involve identifying the domain when the first admin signs up and ensuring its uniqueness within the workspace table. Since numerous workspaces have already configured their settings, we plan to implement this change without causing disruption in a subsequent PR.

image

### CODE REVIEW

**Comment by flvndvd** [comment]
@Duncid if you want to review the copy 🙏?

**Comment by Duncid** [comment]
I feel like the existing copy is already pretty explicit:
image

File: front/pages/api/w/[wId]/index.ts [hunk 1]
```diff
@@ -83,16 +85,27 @@ async function handler(
...
+          name: body.name,
         });
-        owner.name = req.body.name as string;
+        owner.name = body.name;
       } else {
-        await w.update({
-          allowedDomain: req.body.allowedDomain,
-        });
-        owner.allowedDomain = req.body.allowedDomain as string | null;
+        const user = auth.user();
```
Comment (spolu) [inline front/pages/api/w/[wId]/index.ts:94]: This route is expecting a user so you can extract it after line 38 and share the error `if (!user || !owner)`

File: front/pages/api/w/[wId]/index.ts [hunk 2]
```diff
@@ -83,16 +85,27 @@ async function handler(
...
+        if (body.autoAddDomainUsers && user) {
+          const [, userEmailDomain] = user.email.split("@");
```
Comment (spolu) [inline front/pages/api/w/[wId]/index.ts:96]: Should not be possible but who knows... but this does not get the domain if there is 2 `@` in the user.email ?

File: front/pages/api/w/[wId]/index.ts [hunk 3]
```diff
@@ -83,16 +85,27 @@ async function handler(
...
+          if (!isJunkEmailDomain(userEmailDomain)) {
```
Comment (spolu) [inline front/pages/api/w/[wId]/index.ts]: We should error if we don't update no?

File: front/pages/w/[wId]/members/index.tsx [hunk 4]
```diff
@@ -608,57 +615,44 @@ function InviteSettingsModal({
...
-            onChange={(e) => {
-              setDomainInput(e);
-              setAllowedDomainError("");
-            }}
+        <p>
+          Should new users with the whitelisted email domain{" "}
+          <span className="font-bold">
+            {owner.allowedDomain ?? userEmailDomain}
+          </span>{" "}
+          be added to your workspace automatically?
```
Comment (spolu) [inline front/pages/w/[wId]/members/index.tsx:638]: This is not true yet right? they still need the link?

File: front/pages/api/w/[wId]/index.ts (hunk 2 above)
Comment (flvndvd) [reply front/pages/api/w/[wId]/index.ts:96]: We only support signing up through OAuth flows today, meaning that the email we have here is coming from either Google and GitHub. 

As this part won't live here for long, we will move it to the Workspace creating, maybe I can enforce a stricter check on the email when working on this subsequent PR, WDYT?

File: front/pages/w/[wId]/members/index.tsx (hunk 4 above)
Comment (flvndvd) [reply front/pages/w/[wId]/members/index.tsx]: Updated, I used an hardcoded boolean to gate those changes until the sign up flow is being updated top display the old copy.

**Review by spolu — APPROVED** [review]

**Review by philipperolet — APPROVED** [review]
Top :fire:

On the follow-up steps, there is a question around unicity of workspace domain. We can:
 - enforce "only workspace per domain";
 - enforce "multiple workspace per domain, but only one workspace with auto-sign for the domain"
 
The first one is ~ cleaner, the second allows multiple workspaces for the purpose of permissions handling, as a workaround until  we have data permission management for dust (far in the future probably). I'd favor the second option, leaving doors open

**Pull request merged**